src/network/webtask.cpp

```diff
@@ -263,60 +267,64 @@ void WebTask::timerEvent(QTimerEvent* event) {
     VERIFY_OR_DEBUG_ASSERT(m_state == State::Pending) {
         return;
     }
-    VERIFY_OR_DEBUG_ASSERT(m_pendingNetworkReply) {
+    auto* const pPendingNetworkReply = m_pendingNetworkReplyWeakPtr.data();
+    VERIFY_OR_DEBUG_ASSERT(pPendingNetworkReply) {
         return;
     }
-    if (m_pendingNetworkReply->isFinished()) {
+    if (pPendingNetworkReply->isFinished()) {
         // Nothing to do
     }
     kLogger.info()
             << this
             << "Aborting after timed out";
-    DEBUG_ASSERT(m_pendingNetworkReply->isRunning());
-    m_pendingNetworkReply->abort();
-    // Aborting the network reply might finish it
-    // immediately. It will be destroyed when handling
-    // the finished signal, i.e. m_pendingNetworkReply
-    // could be nullptr here!
+    DEBUG_ASSERT(pPendingNetworkReply->isRunning());
+    pPendingNetworkReply->abort();
 }
 
 void WebTask::slotNetworkReplyFinished() {
     DEBUG_ASSERT_QOBJECT_THREAD_AFFINITY(this);
 
-    auto* const finishedNetworkReply = qobject_cast<QNetworkReply*>(sender());
-    VERIFY_OR_DEBUG_ASSERT(finishedNetworkReply) {
+    auto* const pFinishedNetworkReply = qobject_cast<QNetworkReply*>(sender());
+    VERIFY_OR_DEBUG_ASSERT(pFinishedNetworkReply) {
         return;
     }
-    finishedNetworkReply->deleteLater();
+    const auto deleteFinishedNetworkReply = ScopedDeleteLater(pFinishedNetworkReply);
+
     if (kLogger.debugEnabled()) {
-        if (finishedNetworkReply->url() == finishedNetworkReply->request().url()) {
+        if (pFinishedNetworkReply->url() == pFinishedNetworkReply->request().url()) {
             kLogger.debug()
                     << this
                     << "Received reply for request"
-                    << finishedNetworkReply->url();
+                    << pFinishedNetworkReply->url();
         } else {
             // Redirected
             kLogger.debug()
                     << this
                     << "Received reply for redirected request"
-                    << finishedNetworkReply->request().url()
+                    << pFinishedNetworkReply->request().url()
                     << "->"
-                    << finishedNetworkReply->url();
+                    << pFinishedNetworkReply->url();
         }
     }
 
-    if (!m_pendingNetworkReply) {
+    // Check correlation between pending and finished reply
+    auto* const pPendingNetworkReply = m_pendingNetworkReplyWeakPtr.data();
+    if (!pPendingNetworkReply) {
```

Comment:
We can already assert here, because the deleteLater() call is the only source for the m_pendingNetworkReplyWeakPtr becoming null.  If it is already null or in general not m_pendingNetworkReply == finishedNetworkReply something bad has happend.

IMHO the Assert below (Line 321) is sufficient.